.github/workflows/e2e-tests.yml

```
@@ -66,32 +66,40 @@ jobs:
         with:
             node-version: lts/*
 
+      - name: Cache Node.js dependencies
+        id: cache-node
+        uses: actions/cache@v4
+        with:
+            path: e2e/node_modules
+            key: ${{ runner.os }}-node-${{ hashFiles('e2e/package-lock.json') }}
+
       - name: Install dependencies in ./e2e to be able to run `playwright merge-reports`
+        if: steps.cache-node.outputs.cache-hit != 'true'
         run: make e2e-setup-ci
 
       - name: Download blob reports from GitHub Actions Artifacts
         uses: actions/download-artifact@v4
         with:
-          path: all-blob-reports
+          path: ./e2e/blob-report
           pattern: blob-report-shard-*
           merge-multiple: true
 
       - name: Verify downloaded artifacts
         run: |
-          echo "Contents of all-blob-reports after download:"
-          ls -R all-blob-reports
+          echo "Contents of ./e2e/blob-report after download:"
+          ls -R ./e2e/blob-report
 
-      - name: Merge into HTML report
-        run: npx playwright merge-reports --reporter html ./all-blob-reports
+      - name: Merge into HTML report using Makefile
```

nit: write concisely and to the main point
(the fact that it's an html report and that it's using a make command is not really important info)
```suggestion
      - name: Merge reports
```